.github/workflows/pr_and_main.yml

```diff
@@ -77,6 +77,17 @@ jobs:
         with:
           key: ${{ matrix.os }}
 
+      - name: Cache conan setup
+        id: conan-cache-key
+        run: |
+          echo "::set-output name=key::$(/bin/date -u "+%Y%m%d")"
+          echo "::set-output name=path::$(conan config home)"
+      - name: Cache conan
+        uses: actions/cache@v1
+        with:
+          path: ${{ steps.conan-cache-key.outputs.path }}
+          key: conan-${{ matrix.os }}-${{ hashFiles('lib/conanfile.txt') }}-${{ steps.conan-cache-key.outputs.key }}
```

Comment:
How would this work when dependencies are defined in `CMakeLists.txt`?

Reply:
Two options:
- Hash the CMakeLists.txt
- Just don't hash anything. In the worst case the cache will be useless for a day, but conan should just build the new dependencies as required, despite old ones in the cache?

Reply:
How is the hash used? How would the 2nd one ever produce a cache hit?

Reply:
The cache would simply be identified by date, that is, as long as we do more than one build per day we should get speedups.

Reply:
Ok